pkg/command/context.go

```diff
@@ -1438,6 +1448,9 @@ func displayContextListOutputWithDynamicColumns(cfg *configtypes.ClientConfig, w
 		dynamicColumns = append(dynamicColumns, "ClusterGroup")
 	}
 	if showAllColumns {
+		if tanzuContextExists {
+			dynamicColumns = append(dynamicColumns, "ProjectID")
```

Comment:
Instead of this extra `if` you can add it at line 1448 directly.  Notice how `ClusterGroup` is an optional column just like this new `ProjectID` column

Reply:
`ProjectID` column is different from `ClusterGroup`. We don't want to show `ProjectID` when user executes `tanzu context list` but only show when executed with `--wide` option.

Reply:
Oh, I thought that was automatic, but I now see that you are right.
Thanks for pushing back 😄